docker/Dockerfile

```diff
@@ -0,0 +1,20 @@
+# Use Alpine as the base image
+FROM alpine:latest
+
+# Install required packages
+RUN apk add --no-cache bash zsh git gcc python3-dev musl-dev linux-headers py3-pip ffmpeg htop screen figlet
+
+# Clone the pikaraoke repository
+RUN git clone -b PiKaraoke-Docker https://github.com/honestlai/pikaraoke.git /pikaraoke
```

Comment:
Does this need to clone a github repo at all? Could it simply copy the contents of ../ ? That way it is always buidling the image from the current code.

Reply:
I messed around with this and I think it can work, but you need to .dockerignore the .venv directory

Reply:
It does, it's how docker containers stay up to date. In a production state the containers are generally sent to docker hub (much liike github), where there are versions that hold the contents of specific versions and base dependencies all within a mostly platform agnostic portable environment.

When a new build comes out from https://github.com/vicwomg/pikaraoke or my branch for instance, than I would build a new container, tag it with whatever syntax I'm using (still learning how proper container version syntax works), and update it on docker hub.

This way, for end users or sysadmins. If they find the :latest build that comes out, has some weird bug, or isn't compatible with an SQL plugin they're using (like my boss recently), the pull for the production environment can go backwards to something like vicwomg/pikaraoke-docker:1.12 on docker hub to satisfy what was working previously, reseting the ephemeral application layer of things, while the volume(s) (the users data and configs), remain and are unaffected by the downgrading of versions.

Reply:
Apologies if you know this already... making some assumptions here

Reply:
I'm new to docker so this is all very educational and def. makes sense. Thanks for the crash course. I'll mull this all over

Reply:
No worries... it's taken me years to wrap my head around docker... sometimes I feel like some kind of tech evangelist when I come across talented tech folk who have never used docker

"Have you heard of the good word from our lord and savior Docker?" 😅

I'm kubernetes and ansible folks feel the same way.

Reply:
Going to go ahead and mark this resolved